pkg/cortex/modules.go

```diff
@@ -421,6 +425,83 @@ func (t *Cortex) initTableManager(cfg *Config) error {
 	return nil
 }
 
+func (t *Cortex) initSingleTableManager(cfg *Config) error {
+	if cfg.Storage.Engine == storage.StorageEngineTSDB {
+		return nil // table manager isn't used in v2
+	}
+
+	err := cfg.Schema.Load()
+	if err != nil {
+		return err
+	}
+
+	// Assume the newest config is the one to use
+	lastConfig := &cfg.Schema.Configs[len(cfg.Schema.Configs)-1]
+
+	if (cfg.TableManager.ChunkTables.WriteScale.Enabled ||
+		cfg.TableManager.IndexTables.WriteScale.Enabled ||
+		cfg.TableManager.ChunkTables.InactiveWriteScale.Enabled ||
+		cfg.TableManager.IndexTables.InactiveWriteScale.Enabled ||
+		cfg.TableManager.ChunkTables.ReadScale.Enabled ||
+		cfg.TableManager.IndexTables.ReadScale.Enabled ||
+		cfg.TableManager.ChunkTables.InactiveReadScale.Enabled ||
+		cfg.TableManager.IndexTables.InactiveReadScale.Enabled) &&
+		(cfg.Storage.AWSStorageConfig.ApplicationAutoScaling.URL == nil && cfg.Storage.AWSStorageConfig.Metrics.URL == "") {
+		level.Error(util.Logger).Log("msg", "WriteScale is enabled but no ApplicationAutoScaling or Metrics URL has been provided")
+		os.Exit(1)
+	}
+
+	tableClient, err := storage.NewTableClient(lastConfig.IndexType, cfg.Storage)
+	if err != nil {
+		return err
+	}
+
+	bucketClient, err := storage.NewBucketClient(cfg.Storage)
+	util.CheckFatal("initializing bucket client", err)
+
+	t.tableManager, err = chunk.NewTableManager(cfg.TableManager, cfg.Schema, cfg.Ingester.MaxChunkAge, tableClient, bucketClient)
+	if err != nil {
+		return err
+	}
+
+	//This method gets the table manager owner by default table manager
+	//owner is lowest token holding cortex instance
+	t.tableManager.Ring = t.ring
+	tableManagerOwner, err := t.tableManager.SyncTableManagerOwnership(cfg.Ingester.LifecyclerConfig)
```

Comment:
I wouldn't run this here.

I recommend running this at the start of the `loop` function within the table manager: https://github.com/cortexproject/cortex/blob/46b99cf48071cb4674576beabfa1ec7216a2f1cd/pkg/chunk/table_manager.go#L167

Every binary should start their TableManager. The loop will only become active, if their embedded Ingester has the lowest token. This means you don't need the loop below.